solpolpy/core.py

```diff
@@ -158,6 +168,63 @@ def check_alpha_requirement(path: t.List[str]) -> bool:
     return requires_alpha
 
 
+def generate_imax_matrix(arrayshape) -> np.ndarray:
+    """
+    Define an A matrix with which to convert MZP^ (camera coords) = A x MZP (solar coords)
+
+    Parameters
+    -------
+    arrayshape
+        Defined input WCS array shape for matrix generation
+
+    Returns
+    -------
+    ndarray
+        Output A matrix used in converting between camera coordinates and solar coordinates
+
+    """
+
+    # Ideal MZP wrt Solar North
+    thmzp = [-60, 0, 60] * u.degree
+
+    long_arr, lat_arr = np.meshgrid(np.linspace(-20, 20, arrayshape[0]), np.linspace(-20, 20, arrayshape[1]))
+
+    # Foreshortening (IMAX) effect on polarizer angle
+    phi_m = np.arctan2(np.tan(thmzp[0]) * np.cos(long_arr * u.degree), np.cos(lat_arr * u.degree)).to(u.degree)
+    phi_z = np.arctan2(np.tan(thmzp[1]) * np.cos(long_arr * u.degree), np.cos(lat_arr * u.degree)).to(u.degree)
+    phi_p = np.arctan2(np.tan(thmzp[2]) * np.cos(long_arr * u.degree), np.cos(lat_arr * u.degree)).to(u.degree)
+
+    phi = np.stack([phi_m, phi_z, phi_p])
+
+    # Define the A matrix
+    mat_a = np.empty((arrayshape[0], arrayshape[1], 3, 3))
+
+    for i in range(3):
+        for j in range(3):
+            mat_a[:, :, i, j] = (4 * np.cos(phi[i] - thmzp[j]) ** 2 - 1) / 3
+
+    return mat_a
+
+
+def resolve_imax_effect(input_data: NDCollection) -> NDCollection:
+
+    for i, key in enumerate(input_data.keys()):
+        if i == 0:
```

Comment:
It's not immediately obvious to me what's happening. Please add a comment on why you're checking only the first key. Also, check that order doesn't matter. A dictionary does not guarantee to provide .keys() in a specific order.

Reply:
Still need to clarify/resolve this section.